fiaas_skipper/deploy/deploy.py

```diff
@@ -40,9 +40,12 @@ def deploy(self, namespaces=None):
             if namespaces and deployment_config.namespace not in namespaces:
                 continue
             channel = self._release_channel_factory(deployment_config.name, deployment_config.tag)
-            self._deploy(deployment_config, channel)
-            if requires_bootstrap(deployment_config):
-                self._bootstrap(deployment_config, channel, self._spec_config)
+            try:
```

Comment:
We are starting a new thread by calling `Deployer.deploy`, and since there are other things going on in this method that can also fail, we also need to add a catch-all exception handler at the top level. It probably makes sense to do this where we start the thread since this type of error handling might not make sense if the method is called from a different context. Thoughts?

Reply:
Two different concerns. This deals with handling errors for each namespace, so that we can continue with the next one in the list. You are talking about a catch-all at the top level, so that we can log exceptions before the thread dies. Ideally we would like both, but this PR solves the problem of continuing with the next namespace after a failure.

On the other hand, we should probably include the channel factory code, since that also does a remote call. I'll write new PRs for each issue.